nemo/collections/tts/torch/tts_tokenizers.py

```diff
@@ -525,3 +529,127 @@ def set_phone_prob(self, prob):
         finally:
             if hasattr(self.g2p, "phoneme_probability"):
                 self.g2p.phoneme_probability = self.phoneme_probability
+
+
+@experimental
+class PhonemizerTokenizer(IPATokenizer):
+    # fmt: off
+    # Common English punctiation. This is extended with language-specific punctuation during initialization.
+    # This can be overridden using "non_default_punct_list"
+    PUNCT_LIST = (
+        ',', '.', '!', '?', '-',
+        ':', ';', '/', '"', '(',
+        ')', '[', ']', '{', '}',
+    )
+    ESPEAK_SYMBOLS = ('ˈ', 'ˌ', "'", '-', 'ː')
+    # This is a common alphabet containing all IPA phonemes produced by eSpeak for English, Spanish, and German.
+    # This can be overridden using "non_default_chars"
+    CHARS = (
+        'a', 'b', 'd', 'e', 'f', 'h', 'i', 'j', 'k', 'l',
+        'm', 'n', 'o', 'p', 'r', 's', 't', 'u', 'w', 'x',
+        'ð', 'ŋ', 'ə', 'ɛ', 'ɟ', 'ɡ', 'ɣ', 'ɪ', 'ɲ', 'ɾ',
+        'ʃ', 'ʊ', 'ʎ', 'ʒ', 'ʝ', 'β', 'θ'
+    )
```

Comment:
This will get messy as the number of languages increase. Maybe split into a another folder, where you list each language's vocab and add each new langauge's phones to a set upon class init. Although, that brings up the question of how you are going to add a common ipa dict for all languages.

Reply:
@XuesongYang Do you have any inputs on this?

Reply:
I will create a separate file with the phones & punctuation for each language more clearly enumerated.

I don't expect there to be a common ipa dict for all languages, but the default is just a blank dictionary (with a few toy entries to show what the format/syntax is) that can be used & added to before something better is available. For multi-lingual TTS I am guessing each language will still have their own independent pipelines and there will be a high-level tokenizer that takes (text, language) as input and sends it to the language-specific tokenizers.

What is the difference here between a languages "vocab" and its "phones"?

Reply:
In your example, we should should a set of phones for each language, but you already add a list of punctuation. Phones are strictly a subset of an language's vocab.